### ISSUE TITLE
Flaky test: CheckShutdownEventLogMessage - StdOut Logging

### ISSUE THREAD

**Issue by ryanbrandenburg** [aspnet/IISIntegration#1100]
This test [fails](http://aspnetci/viewLog.html?buildId=515544&buildTypeId=Releases_22xPublic_Win2012) occasionally with the following error:
```
Assert.Contains() Failure
Not found: (filter expression)
In value:  ConcurrentQueue<WriteContext> [WriteContext { EventId = 0, Exception = null, Formatter = Func`3 { ... }, LoggerName = "Microsoft.AspNetCore.Server.IntegrationTesting.IIS"..., LogLevel = Information, ... }, WriteContext { EventId = 0, Exception = null, Formatter = Func`3 { ... }, LoggerName = "Microsoft.AspNetCore.Server.IntegrationTesting.IIS"..., LogLevel = Information, ... }, WriteContext { EventId = 0, Exception = null, Formatter = Func`3 { ... }, LoggerName = "Microsoft.AspNetCore.Server.IntegrationTesting.IIS"..., LogLevel = Information, ... }, WriteContext { EventId = 0, Exception = null, Formatter = Func`3 { ... }, LoggerName = "Microsoft.AspNetCore.Server.IntegrationTesting.IIS"..., LogLevel = Information, ... }, WriteContext { EventId = 0, Exception = null, Formatter = Func`3 { ... }, LoggerName = "Microsoft.AspNetCore.Server.IntegrationTesting.IIS"..., LogLevel = Information, ... }, ...]
   at Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests.EventLogHelpers.VerifyEventLogEvent(ITestSink testSink, String expectedRegexMatchString) in /_/test/Common.FunctionalTests/Utilities/EventLogHelpers.cs:line 16
   at Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests.EventLogTests.CheckShutdownEventLogMessage() in /_/test/Common.FunctionalTests/Inprocess/EventLogTests.cs:line 34
--- End of stack trace from previous location where exception was thrown ---
------- Stdout: -------
| [0.001s] TestLifetime Information: Starting test CheckShutdownEventLogMessage at 2018-07-25T22:43:55
| [0.004s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: Deploying [Variation] :: ServerType=IIS, Runtime=CoreClr, Arch=x64, BaseUrlHint=, Publish=True
| [0.004s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: Removing environment variable ASPNETCORE_ENVIRONMENT
| [0.004s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: Executing command dotnet publish  --output "C:\b\t\buildTmp\c896b426-0c63-42e6-990c-613563f25de9" --framework netcoreapp2.2 --configuration Release --no-restore -p:VerifyMatchingImplicitPackageVersion=false 
| [0.218s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: dotnet-publish stdout: Microsoft (R) Build Engine version 15.7.179.6572 for .NET Core
| [0.219s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: dotnet-publish stdout: Copyright (C) Microsoft Corporation. All rights reserved.
| [1.055s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: dotnet-publish stdout:   TestTasks -> C:\b\w\33bdfc1cae7b2a38\modules\IISIntegration\test\TestTasks\bin\Release\netcoreapp2.2\TestTasks.dll
| [1.137s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: dotnet-publish stdout:   Microsoft.AspNetCore.Server.IIS -> C:\b\w\33bdfc1cae7b2a38\modules\IISIntegration\src\Microsoft.AspNetCore.Server.IIS\bin\Release\netstandard2.0\Microsoft.AspNetCore.Server.IIS.dll
| [1.488s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: dotnet-publish stdout:   InProcessWebSite -> C:\b\w\33bdfc1cae7b2a38\modules\IISIntegration\test\WebSites\InProcessWebSite\bin\Release\netcoreapp2.2\InProcessWebSite.dll
| [2.212s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: dotnet-publish stdout:   InProcessWebSite -> C:\b\t\buildTmp\c896b426-0c63-42e6-990c-613563f25de9\
| [2.259s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: dotnet publish finished with exit code : 0
| [2.261s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: Backed up C:\Windows\system32\inetsrv\config\applicationhost.config to C:\Windows\system32\inetsrv\config\applicationhost.config.HttpTestSite20180725154355backup
| [2.268s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: Copied contents of IIS.config to C:\Windows\system32\inetsrv\config\applicationhost.config
| [2.276s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: Could not add environment variables to worker process: Filename: 
|                                                                                      Error: Unrecognized element 'environmentVariables'
| [2.277s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: Configured AppPool HttpTestSite20180725154355Pool
| [2.277s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: Configured Site HttpTestSite20180725154355 with AppPool HttpTestSite20180725154355Pool
| [2.277s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: Could not find AspNetCoreModuleV2 section in global modules; creating section.
| [2.278s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: Could not find AspNetCoreModuleV2 section in modules; creating section.
| [2.324s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: Site HttpTestSite20180725154355 has started.
| [2.324s] Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer Information: Successfully finished IIS application directory setup.
| [2.324s] HttpTestSite Debug: Method: GET, RequestUri: 'http://localhost:50658/HelloWorld', Version: 2.0, Content: <null>, Headers:
|                              {
|                              }
| [3.863s] HttpTestSite Debug: StatusCode: 200, ReasonPhrase: 'OK', Version: 1.1, Content: System.Net.Http.HttpConnection+HttpConnectionResponseContent, Headers:
|                              {
|                                Transfer-Encoding: chunked
|                                Server: Microsoft-IIS/8.5
|                                Date: Wed, 25 Jul 2018 22:43:59 GMT
|                              }
| [3.870s] HttpTestSite Debug: Hello World
| [3.873s] Microsoft.AspNetCore.Server
...
```

CC @Eilon,@muratg,@mkArtakMSFT

This issue was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
Please use this workflow to address this flaky test issue, including checking applicable checkboxes and filling in the applicable "TODO" entries:

* Is this test failure caused by product code flakiness? (Either this product, or another product this test depends on.)
  * [ ] File a bug against the product (TODO: Link to other bug)
  * Is it possible to change the test to avoid the flakiness?
    * Yes? Go to the "Change the test!" section.
    * No?
      * [ ] Disable the test (TODO: Link to PR/commit)
      * [ ] Wait for other bug to be resolved
      * [ ] Wait for us to get build that has the fix
      * [ ] Re-enable our test (TODO: Link to PR/commit)
      * [ ] Close this bug

* Is it that the test itself is flaky? This includes external transient problems (e.g. remote server problems, file system race condition, etc.)
  * Is there is a way to change our test to avoid this flakiness?
    * Yes? Change the test!
      * [ ] Change the test to avoid the flakiness, for example by using a different test strategy, or by adding retries w/ timeouts (TODO: Link to PR/commit)
      * [ ] Run the test 100 times locally as a sanity check.
      * [ ] Close this bug
    * No?
      * [ ] Delete the test because flaky tests are not useful (TODO: Link to PR/commit)

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
CheckShutdownEventLogMessage [failed again](http://aspnetci/viewLog.html?buildId=515925&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by Eilon** [comment]
@jkotalik - please take a look at this flaky test ASAP (or re-assign).

**Comment by jkotalik** [comment]
There are a slew of flaky tests that I'm focusing on now. This will be resolved ASAP.

**Comment by ryanbrandenburg** [comment]
CheckShutdownEventLogMessage [failed again](http://aspnetci/viewLog.html?buildId=516438&buildTypeId=XPlat_Windows_Win10Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
CheckShutdownEventLogMessage [failed again](http://aspnetci/viewLog.html?buildId=516957&buildTypeId=Releases_22xPublic_Win2012).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by pakrym** [comment]
https://ci.appveyor.com/project/aspnetci/iisintegration/build/1.0.1692

**Comment by ryanbrandenburg** [comment]
CheckShutdownEventLogMessage [failed again](http://aspnetci/viewLog.html?buildId=517599&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
CheckShutdownEventLogMessage [failed again](http://aspnetci/viewLog.html?buildId=518163&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
InvalidProcessPath_ExpectServerError [failed with about the same error](http://aspnetci/viewLog.html?buildId=520631&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by mikeharder** [comment]
We believe this issue will be addressed by https://github.com/aspnet/IISIntegration/pull/1154

**Comment by pakrym** [comment]
https://dotnet.visualstudio.com/public/_build/results?buildId=13193&view=logs

**Comment by pakrym** [comment]
I don't think this would get fixed by #1154 anymore. I think the issue is that we have to call StopServer before checking log output.
